src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -7484,7 +7484,7 @@ interface CorporationInfo {
   dividendTax: number;
   /** Your earnings as a shareholder per second this cycle */
   dividendEarnings: number;
-  /** State of the corporation. Possible states are START, PURCHASE, PRODUCTION, EXPORT, SALE. */
+  /** The next state to be processed. Possible states are START, PURCHASE, PRODUCTION, EXPORT, SALE. */
```

Comment:
Since this is a comment, we have more room for details. I'd add a sentence like: "I.e. when the state is PURCHASE, it means purchasing will occur during the next state transition."

Reply:
Agreed, also I think this is the main place to make the clarification, I'm not sure the edits in the UI were necessary (were people confused by the UI description, or mostly just by the state in scripts?)

Also `npm run doc` will be needed to generate the new .md from this doc change

Reply:
I really think clarity here needs to be supported by more comprehensive information update, and the changes being discussed are at best trading one confusion for another. 
"Waiting to purchase" is state, "next state selling" is [e.g.] nextState. The property used in game is `.state`. At least it still strikes me as confusing and less descriptive, for what it's worth.
> I think this is the main place to make the clarification

In the description for ns.corporation.getCorporation().state? :P IMO, if it's not clear from the Divisions subpage, players aren't finding it.
![image](https://github.com/bitburner-official/bitburner-src/assets/141260118/76538efa-9070-4b09-8e19-b31825444ee6)
currently one of the few pages with italics and cool formula. cool. So why not add a Gantt chart or something that makes it super clear the order of operations and player's options during each state and the effects of each transition; or a process flow chart [presumably as a helper modal, as implied previously]. If it's causing confusion and has such fine detail, an extra sentence buried in a property is likely insufficient.

Just trying to avoid kicking a problem down the road and imagining alternatives.

Reply:
> Since this is a comment, we have more room for details. I'd add a sentence like: "I.e. when the state is PURCHASE, it means purchasing will occur during the next state transition."

![image](https://github.com/bitburner-official/bitburner-src/assets/115591472/b2a1e006-ffe8-484b-85b3-4c81a714e586)